scipy/sparse/linalg/eigen/lobpcg/lobpcg.py

```diff
@@ -455,11 +467,21 @@ def lobpcg(A, X,
         # Apply constraints to the preconditioned residuals.
         if blockVectorY is not None:
             _applyConstraints(activeBlockVectorR,
-                              gramYBY, blockVectorBY, blockVectorY)
+                              YBY, blockVectorBY, blockVectorY)
 
         ##
-        # B-orthonormalize the preconditioned residuals.
+        # B-orthogonalize the preconditioned residuals to X.
+        if B is not None:
+            activeBlockVectorR = activeBlockVectorR - np.matmul(blockVectorX,
```

Comment:
You can use @ for matrix multiplication since we dropped python 2 support

Reply:
> 
> 
> You can use @ for matrix multiplication since we dropped python 2 support

There are lots of matrix multiplication in the code to change into @. I have already made too many changes for my comfort level in this PR. Can we keep @ for the next PR, please?

Reply:
Ah yes. It was meant as a suggestion but shouldn't have been inside the review. Please take it as a reminder. Not a blocker.